sympy/combinatorics/homomorphisms.py

```diff
@@ -505,13 +502,18 @@ def find_homomorphism(G, H, injective=True, surjective=True, compute=True, all=F
             raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
         _H, h_isomorphism = H._to_perm_group()
 
-    if injective:
-        if (g_order != h_order) or not G.is_abelian and H.is_abelian:
+    if injective and surjective:
+        if (g_order != h_order) or (G.is_abelian != H.is_abelian):
```

Comment:
The condition on order can be split up and each part put under either `if injective` or `if surjective`. I mentioned this the first time I suggested these keywords: you can't have an injective homomorphism if `g_order` doesn't divide `h_order` (just by Lagrange's theorem), and you can't have a surjective one if `h_order` doesn't divide `g_order`. When you have both `injective` and `surjective` set to `True`, these two conditions will automatically give you `g_order != h_order`